pyproject.toml

```diff
@@ -33,9 +33,13 @@ gunicorn = "^20.1"
 python-dateutil = "^2.8"
 pyyaml = "^6"
 h5py = { version = "^3.6", optional = true }
-apache-beam = { extras = ["gcp"], version = "^2.37", optional = true }
+apache-beam = { version = "^2.37", optional = true }
 twined = "^0.5.0"
-packaging = "^21.3"
+packaging = ">=20.4,<22"
+cachetools = ">=3.1.0,<5"
+google-apitools = ">=0.5.31,<0.5.32"
+google-auth-httplib2 = ">=0.1.0,<0.2.0"
+google-cloud-datastore = ">=1.8.0,<2"
```

Comment:
I'd add comments about why and which ones are as a result of the apache beam issues/limitations. In a related service I did something like:
```
        "octue==0.27.0", # actually I need octue[dataflow] which installs apache-beam[gcp]
        "apache-beam==2.40.0",  # actually I need apache-beam[gcp] but see https://github.com/apache/beam/issues/21019
        "cachetools>=3.1.0,<5",  # from apache-beam[gcp] - update if you update apache-beam
        "google-apitools>=0.5.31,<0.5.32",  # from apache-beam[gcp] - update if you update apache-beam
        "google-auth>=1.18.0,<3",  # from apache-beam[gcp] - update if you update apache-beam
        "google-auth-httplib2>=0.1.0,<0.2.0",  # from apache-beam[gcp] - update if you update  apache-beam
        "google-cloud-datastore>=1.8.0,<2",  # from apache-beam[gcp] - update if you update apache-beam
        "google-cloud-pubsub>=2.1.0,<3",  # from apache-beam[gcp] - update if you update apache-beam
        "google-cloud-pubsublite>=1.2.0,<2",  # from apache-beam[gcp] - update if you update apache-beam
        # ... and I don't bother installing the unused test dependencies which include tight bigquery and spanner constraints
    ]
    ```